cypher-shell/src/integration-test/java/org/neo4j/shell/MainIntegrationTest.java

```diff
@@ -1,64 +1,136 @@
-        String inputString = String.format( "neo4j%nneo%n" );
-        InputStream inputStream = new ByteArrayInputStream(inputString.getBytes());
-
         ByteArrayOutputStream baos = new ByteArrayOutputStream();
-        PrintStream ps = new PrintStream(baos);
-
-        Main main = new Main(inputStream, ps);
+        Main main = getMain( baos );
 
         CliArgs cliArgs = new CliArgs();
-        cliArgs.setUsername("", "");
+        cliArgs.setUsername( "", "" );
         cliArgs.setPassword( "", "" );
 
-        Logger logger = new AnsiLogger(cliArgs.getDebugMode());
-        PrettyConfig prettyConfig = new PrettyConfig(cliArgs);
-        ConnectionConfig connectionConfig = new ConnectionConfig(
-                cliArgs.getScheme(),
-                cliArgs.getHost(),
-                cliArgs.getPort(),
-                cliArgs.getUsername(),
-                cliArgs.getPassword(),
-                cliArgs.getEncryption(),
-                cliArgs.getDatabase());
-
-        CypherShell shell = new CypherShell(logger, prettyConfig);
+        ShellAndConnection sac = getShell( cliArgs );
+        CypherShell shell = sac.shell;
+        ConnectionConfig connectionConfig = sac.connectionConfig;
 
         // when
-        assertEquals("", connectionConfig.username());
-        assertEquals("", connectionConfig.password());
+        assertEquals( "", connectionConfig.username() );
+        assertEquals( "", connectionConfig.password() );
 
-        main.connectMaybeInteractively(shell, connectionConfig, true);
+        main.connectMaybeInteractively( shell, connectionConfig, true );
 
         // then
         // should be connected
-        assertTrue(shell.isConnected());
+        assertTrue( shell.isConnected() );
         // should have prompted and set the username and password
-        assertEquals("neo4j", connectionConfig.username());
-        assertEquals("neo", connectionConfig.password());
+        assertEquals( "neo4j", connectionConfig.username() );
+        assertEquals( "neo", connectionConfig.password() );
 
         String out = baos.toString();
         assertEquals( String.format( "username: neo4j%npassword: ***%n" ), out );
     }
+
+    @Test
+    public void wrongPortWithBolt() throws Exception
+    {
+        // given
+        Main main = getMain( new ByteArrayOutputStream() );
+
+        CliArgs cliArgs = new CliArgs();
+        cliArgs.setScheme( "bolt://", "" );
+        cliArgs.setPort( 1234 );
+
+        ShellAndConnection sac = getShell( cliArgs );
+        CypherShell shell = sac.shell;
+        ConnectionConfig connectionConfig = sac.connectionConfig;
+
+        exception.expect( ServiceUnavailableException.class );
+        exception.expectMessage( "Unable to connect to localhost:1234, ensure the database is running and that there is a working network connection to it" );
+        main.connectMaybeInteractively( shell, connectionConfig, true );
+    }
+
+    @Ignore
```

Comment:
It might be useful to have a comment here about what it would take to un-ignore this test, so future work can do that if it appears the blocker is removed.